File: budget-mate-backend/src/main/java/com/budget/mate/services/CardService.java
package com.budget.mate.services;

import com.budget.mate.domain.CardEntity;
import com.budget.mate.dto.CardDto;
import com.budget.mate.dto.CreateCardDto;

import java.util.List;

public interface CardService {
    CardDto addCard(CreateCardDto cardDto);

    Boolean removeCardById(String cardId);

    List<CardDto> findAllMyCards();

    CardDto findCardById(String cardId);

    CardDto updateCard(String cardId, CardDto cardDto);

    CardDto selectCardAsPrimary(String cardId);

    CardEntity findCardEntityById(String cardId);
}

File: budget-mate-frontend/src/app/dashboard/services/card.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {CardDto} from "../../models/card-dto";
import {url} from "../../../environments/environment";
import {CreateCardDto} from "../../models/create-card-dto";

@Injectable({
  providedIn: 'root'
})
export class CardService {

  constructor(private http: HttpClient) {
  }

  public findAllMyCards(): Observable<CardDto[]> {
    const req = `${url}/cards`
    return this.http.get<CardDto[]>(req);
  }

  public updateCardById(cardId:string,cardDto:CardDto): Observable<CardDto> {
    const req = `${url}/cards/${cardId}`
    return this.http.put<CardDto>(req,cardDto);
  }

  public removeCardById(cardId: string): Observable<boolean> {
    const req = `${url}/cards/${cardId}`;
    return this.http.delete<boolean>(req);
  }

  addCard(createCardDto: CreateCardDto): Observable<CardDto> {
    const req = `${url}/cards`;
    return this.http.post<CardDto>(req, createCardDto);
  }
}
